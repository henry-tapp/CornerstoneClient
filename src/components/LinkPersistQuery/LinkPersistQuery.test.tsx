import React from "react";
import { HashRouter, MemoryRouter } from "react-router-dom";

import { render, screen } from "@testing-library/react";

import { LinkPersistQuery } from "./";

it("renders without crashing", () => {
  render(<LinkPersistQuery pathname="">A Link</LinkPersistQuery>, {
    wrapper: HashRouter,
  });
});

it("maintains the query string in the link", () => {
  const initialRoute = {
    pathname: "/somePath",
    search: "?authToken=123&theme=party",
  };
  const { asFragment } = render(
    <MemoryRouter initialEntries={[initialRoute]}>
      <LinkPersistQuery pathname="/somePath">A Link</LinkPersistQuery>
    </MemoryRouter>
  );

  expect(screen.getByText(/A Link/i)).toBeInTheDocument();
  expect(screen.getByRole("link")).toHaveAttribute(
    "href",
    "/somePath?authToken=123&theme=party"
  );
  expect(asFragment()).toMatchSnapshot();
});

it("adds the .active class if the path matches", () => {
  const initialRoute = {
    pathname: "/somePath",
    search: "?authToken=123&theme=party",
  };
  render(
    <MemoryRouter initialEntries={[initialRoute]}>
      <LinkPersistQuery pathname="/somePath">A Link</LinkPersistQuery>
    </MemoryRouter>
  );

  expect(screen.getByText(/A Link/i)).toBeInTheDocument();
  expect(screen.getByRole("link")).toHaveClass("active");
});

it("adds the .active class if the path is a decendant", () => {
  const initialRoute = {
    pathname: "/somePath/achildpath",
    search: "?authToken=123&theme=party",
  };
  render(
    <MemoryRouter initialEntries={[initialRoute]}>
      <LinkPersistQuery pathname="/somePath">A Link</LinkPersistQuery>
    </MemoryRouter>
  );

  expect(screen.getByText(/A Link/i)).toBeInTheDocument();
  expect(screen.getByRole("link")).toHaveClass("active");
});

it("does not add the .active class if the path doesn't match", () => {
  const initialRoute = {
    pathname: "/someOtherPath",
    search: "?authToken=123&theme=party",
  };
  render(
    <MemoryRouter initialEntries={[initialRoute]}>
      <LinkPersistQuery pathname="/somePath">A Link</LinkPersistQuery>
    </MemoryRouter>
  );

  expect(screen.getByText(/A Link/i)).toBeInTheDocument();
  expect(screen.getByRole("link")).not.toHaveClass("active");
});
