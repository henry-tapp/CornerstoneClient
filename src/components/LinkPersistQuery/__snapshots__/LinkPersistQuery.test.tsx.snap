// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`maintains the query string in the link 1`] = `
<DocumentFragment>
  <a
    class="link-persist-query active "
    href="/somePath?authToken=123&theme=party"
  >
    A Link
  </a>
</DocumentFragment>
`;
